### PR TITLE
Item related mappings

### PR DESCRIPTION
**Pull request by DeadlyMC**
- Named several item related mappings.
- `ItemPropertyGetter` -> `ModelPredicateProvider`

Some controversial names:
- `ItemWithMetadata.Types`
- `SPECKLED_MELON` -> `GLISTERING_MELON_SLICE` ? (Its called "speckled_melon" in the registry)
 
Fixes:
- `Block.fromNane` -> `Block.fromName`
- `Blocks.register` -> `Blocks.get` (Similar to `Items.get`)
- `EntityGroup.UNDEFINED` -> `EntityGroup.DEFAULT`
- `EnchantedGoldenAppleItem` -> `GoldenAppleItem`

### CODE REVIEW

**Review by hYdos — APPROVED** [review]

File: mappings/net/minecraft/client/item/ModelPredicateProvider.mapping [hunk 1]
```diff
@@ -0,0 +1,5 @@
+CLASS net/minecraft/class_5014 net/minecraft/client/item/ModelPredicateProvider
+	METHOD method_25625 call (Lnet/minecraft/class_5013;Lnet/minecraft/class_5086;Lnet/minecraft/class_2267;)F
```
Comment (shedaniel) [inline mappings/net/minecraft/client/item/ModelPredicateProvider.mapping:2]: maybe provide?

**Review by shedaniel — APPROVED** [review]

File: mappings/net/minecraft/client/item/ModelPredicateProvider.mapping (hunk 1 above)
Comment (DeadlyMC) [reply mappings/net/minecraft/client/item/ModelPredicateProvider.mapping:2]: Its called `call` in upstream too
https://github.com/FabricMC/yarn/blob/1.16.4/mappings/net/minecraft/client/item/ModelPredicateProvider.mapping

Comment (shedaniel) [reply mappings/net/minecraft/client/item/ModelPredicateProvider.mapping:2]: why does upstream bad

Comment (DeadlyMC) [reply mappings/net/minecraft/client/item/ModelPredicateProvider.mapping:2]: I think the priority is to follow upstream...so cant do anything ig ¯\_(ツ)_/¯

Comment (shedaniel) [reply mappings/net/minecraft/client/item/ModelPredicateProvider.mapping:2]: yep this is ok then

Comment (DeadlyMC) [reply mappings/net/minecraft/client/item/ModelPredicateProvider.mapping:2]: alrighty

**Comment by hYdos** [comment]
Looks ready to merge. merging now

File: mappings/net/minecraft/entity/EntityGroup.mapping [hunk 2]
```diff
@@ -0,0 +1,2 @@
+CLASS net/minecraft/class_2280 net/minecraft/entity/EntityGroup
+	FIELD field_33254 DEFAULT Lnet/minecraft/class_2280;
```
Comment (BoogieMonster1O1) [inline mappings/net/minecraft/entity/EntityGroup.mapping:2]: is this an enum
if it is dont rename it
`genSources` will scream

Comment (DeadlyMC) [reply mappings/net/minecraft/entity/EntityGroup.mapping:2]: Upstream does it too
https://github.com/FabricMC/yarn/blob/20w51a/mappings/net/minecraft/entity/EntityGroup.mapping

**Pull request merged**